src/ResourceManager/Compute/Commands.Compute/Extension/VmssDiskEncryption/GetAzureVmssVMDiskEncryptionStatus.cs

```diff
@@ -175,33 +177,46 @@ private EncryptionStatus GetOsDiskEncryptionStatus(List<DiskInstanceView> disks,
                 : ConvertToEncryptionStatus(status.Code.Replace(AzureVmssDiskEncryptionExtensionContext.EncryptionStateString, ""));
         }
 
-        private EncryptionStatus GetDataDiskEncryptionStatus(List<DiskInstanceView> disks, StorageProfile storage)
+        private EncryptionStatus GetDataDiskEncryptionStatus(string rgName, string vmssName, List<DiskInstanceView> disks, StorageProfile storage)
         {
             if (storage == null || storage.DataDisks == null || storage.DataDisks.Count == 0)
             {
                 return EncryptionStatus.NotMounted;
             }
 
-            InstanceViewStatus status = null;
+            // Data disk does not have disk encryption extension setting.
+
+            var vmssResult = this.VirtualMachineScaleSetClient.Get(rgName, vmssName);
+            if (vmssResult.VirtualMachineProfile == null
+                || vmssResult.VirtualMachineProfile.ExtensionProfile == null
+                || vmssResult.VirtualMachineProfile.ExtensionProfile.Extensions == null
+                || vmssResult.VirtualMachineProfile.ExtensionProfile.Extensions.Count == 0)
+            {
+                return EncryptionStatus.NotEncrypted;
+            }
+
             try
             {
-                var disk = disks.First(e => e.Name.Equals(storage.DataDisks[0].Name));
+                VirtualMachineScaleSetExtension ext = vmssResult.VirtualMachineProfile.ExtensionProfile.Extensions.First(
+                         e => e.Type.Equals(this.ExtensionName));
 
-                if (disk == null)
+                AzureVmssDiskEncryptionExtensionPublicSettings encryptionSettings = JsonConvert.DeserializeObject<AzureVmssDiskEncryptionExtensionPublicSettings>(
+                ext.Settings.ToString());
+                if (encryptionSettings.VolumeType.Equals(AzureVmssDiskEncryptionExtensionContext.VolumeTypeAll, StringComparison.OrdinalIgnoreCase)
+                    || encryptionSettings.VolumeType.Equals(AzureVmssDiskEncryptionExtensionContext.VolumeTypeData, StringComparison.OrdinalIgnoreCase))
                 {
-                    return EncryptionStatus.Unknown;
+                    if (encryptionSettings.EncryptionOperation.Equals(AzureDiskEncryptionExtensionConstants.enableEncryptionOperation, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return EncryptionStatus.Encrypted;
```

Comment:
And extension provisioning state should have succeeded

Reply:
This function returns only data disk encryption status.  The cmdlets gets the extension provisioning state from VMSS VM instance view (line 146).